users/serializers.py

```diff
@@ -2,26 +2,20 @@
 from django.contrib.auth.models import User, Group
 
 
-class GroupSerializer(serializers.Serializer):
+class GroupSerializer(serializers.ModelSerializer):
     class Meta:
         model = Group
         fields = ('name',)
 
 
-class UserSerializer(serializers.Serializer):
-    groups = GroupSerializer(many=True, required=False)
-    username = serializers.EmailField(allow_blank=False)
-    first_name = serializers.CharField(max_length=200, allow_blank=False)
-    last_name = serializers.CharField(max_length=200, allow_blank=False)
-    email = serializers.EmailField(allow_blank=False)
-    password = serializers.CharField(style={'input_type': 'password'}, allow_blank=False)
+class UserSerializer(serializers.ModelSerializer):
+    groups_complete = GroupSerializer(source='groups', read_only=True, many=True)
 
-    def create(self, validated_data):
-        username = validated_data.get('username')
-        first_name = validated_data.get('first_name')
-        last_name = validated_data.get('last_name')
-        email = validated_data.get('email')
-        password = validated_data.get('password')
-        return User.objects.create_user(
-            username=username, first_name=first_name, last_name=last_name,
-            email=email, password=password)
+    class Meta:
+        model = User
+        fields = ('id', 'username', 'password', 'first_name',
+                  'last_name', 'groups', 'groups_complete')
+        extra_kwargs = {
+                        'username': {'required': True},
+                        'password': {'required': True}
+                        }
```

Comment:
If I understand correctly. When a user is created, the API receives the group of the user to be created. But am not sure how saving nested serializers works (I know you can't save nested objects in DRF), I mean you are saving a n object that depends on another. Maybe make so tests trying to create different groups of users.